src/app/app.component.ts

```diff
@@ -88,9 +88,9 @@ export class AppComponent implements OnInit {
   // pythonWorker: Worker | null = null;
 
   constructor(
-    private messageService: MessageService,
-    private storageService: StorageService,
-    private workerService: WorkerService
+    readonly messageService: MessageService,
```

Comment:
private and readonly doesn't have the same purpose and your objects could be both.
```suggestion
    private readonly messageService: MessageService,
```